app/src/courses/courses.service.ts

```
@@ -750,6 +781,74 @@ export class CoursesService {
+        return attempts.sort((a, b) => b.attempt - a.attempt)[0];
+      
+      case AttemptsGradeMethod.HIGHEST:
+        return attempts.reduce((best, current) => {
+          const bestScore = best.score || 0;
+          const currentScore = current.score || 0;
+          return currentScore > bestScore ? current : best;
+        });
+      
+      case AttemptsGradeMethod.AVERAGE:
+        // For average, we need to calculate averages and return a synthetic attempt
+        const totalScore = attempts.reduce((sum, attempt) => sum + (attempt.score || 0), 0);
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)